portal/static/js/main.js

```diff
@@ -4886,8 +4908,12 @@ var tnthAjax = {
             };
         });
     },
-    "getTerms": function(userId, type, sync, callback) {
-        this.sendRequest('/api/user/'+userId+'/tou'+(type && hasValue(type)?('/'+type):''), 'GET', userId, {sync:sync}, function(data) {
+    "getTerms": function(userId, type, sync, callback, params) {
+        if (!params) params = {};
+        var url = "/api/user/{userId}/tou{type}{all}".replace("{userId}", userId)
```

Comment:
This seems kind of fragile (building QS params by concatenation). Are "all" and "type" mutually exclusive?
Would you be able to use `jQuery.param()` to build it instead?

Reply:
@ivan-c Ivan, "all" is a optional querystring parameer and "type" can be optional part of api path - so they can co-exist.  I am thinking we can't use jQuery.param() here - the serialized string from jQuery.param(), i.e. a=blah&b=blah, will not work in this instance, i.e. /api/user/[type][?all=true]